README.md

```diff
@@ -498,6 +498,80 @@ Flag | Notes
 `--https_proxy`      | Needed for package manager or Ubuntu guest and for from github downloaded packages
 `--ftp_proxy`       | Needed for some Perfkit packages
 
+CONFIGURATIONS AND CONFIGURATION OVERRIDES
+==================
+Each benchmark now has an independent configuration which is written in YAML.
+A user config is a way to override this default configuration on a per benchmark
+basis. This allows for much more complex setups than previously possible,
+including running benchmarks across clouds.
+
+A benchmark configuration has a somewhat simple structure. It is essentially
+just a series of nested dictionaries. At the top level, it contains VM groups.
+VM groups are logical groups of homogenous machines. The VM groups hold both a
+"vm_spec" and a "disk_spec" which contain the parameters needed to
+create members of that group. Here is an example of an expanded configuration:
+```
+iperf:
+  vm_groups:
+    iperf_vm_1:
+      vm_count: 1
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          image: ubuntu-14-04
+          zone: us-central1-c
+        AWS:
+          machine_type: m3.medium
+          image: ami-######
+          zone: us-east-1a
+        # Other clouds here...
+    iperf_vm_2:
+      vm_count: 1
+      # This specifies the cloud to use for the group. This allows for
+      benchmark configurations that span clouds.
+      cloud: AWS
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-1
+          image: ubuntu-14-04
+          zone: us-central1-c
+        # Other clouds here...
+      disk_count: 0
+      disk_spec:
+        GCP:
+          disk_size: 500
+          disk_type: standard
+        # Other clouds here...
+```
+
+For a complete list of keys for vm_specs and disk_specs see
+[virtual_machine.BaseVmSpec](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/virtual_machine.py)
+and
+[disk.BaseDiskSpec](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/disk.py)
+and their derived classes.
+
+User configs are applied on top of the existing default config and can be
+specified in two ways. The first is by supplying a config file via the
+--benchmark_config_file flag. The second is by specifying a single setting to
+override via the `--config_override flag`.
+
+A user config file only needs to specify the settings which it is intended to
+override. For example if the only thing you want to do is change the number of
+VMs in a group, this config is sufficient:
+```
+cluster_boot:
+  vm_groups:
+    default:
+      vm_count: 100
+```
+You can achieve the same effect by specifying the `config_override` flag.
+The value of the flag should be a path within the YAML (with keys delimited by
+periods), an equals sign, and finally the new value:
+```
+--config_override=cluster_boot.vm_groups.default.vm_count=100
```

Comment:
Done. Yes - flags are the most obvious case - also specialized attributes of vm_specs or disk_specs which weren't made explicit in the default spec (e.g. the default GCP vm_spec doesn't specify num_local_ssds, but the override does).